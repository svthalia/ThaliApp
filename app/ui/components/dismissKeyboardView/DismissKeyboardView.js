import React from 'react';
import PropTypes from 'prop-types';
import {
  View, Keyboard, TouchableWithoutFeedback, StyleSheet, ViewPropTypes,
} from 'react-native';

const DismissKeyboardView = (props) => (
  <TouchableWithoutFeedback
    onPress={Keyboard.dismiss}
    accessible={false}
    style={props.style}
  >
    <View
      style={props.contentStyle}
    >
      {props.children}
    </View>
  </TouchableWithoutFeedback>
);

DismissKeyboardView.propTypes = {
  children: PropTypes.node.isRequired,
  style: ViewPropTypes.style,
  contentStyle: ViewPropTypes.style,
};

const styles = StyleSheet.create({
});

DismissKeyboardView.defaultProps = {
  style: styles,
  contentStyle: styles,
};

export default DismissKeyboardView;
