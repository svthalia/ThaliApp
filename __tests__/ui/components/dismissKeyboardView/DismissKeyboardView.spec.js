import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import DismissKeyboardView from '../../../../app/ui/components/dismissKeyboardView/DismissKeyboardView';

describe('DismissKeyboardView component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <DismissKeyboardView>
          <Text>Text</Text>
        </DismissKeyboardView>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
