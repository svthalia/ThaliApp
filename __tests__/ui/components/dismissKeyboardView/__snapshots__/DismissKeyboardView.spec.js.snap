// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DismissKeyboardView component renders correctly 1`] = `
<View
  accessible={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
>
  <Text>
    Text
  </Text>
</View>
`;
